src/lib/utils/index.ts

```diff
@@ -19,22 +27,38 @@ const uploadApiEndpoint = `https://zns-backend.netlify.app/.netlify/functions/up
  * @param params Metadata parameters
  * @returns URI to the created Metadata
  */
-
-export const createDomainMetadata = async (params: DomainMetadataParams) => {
-	// upload image to http backend
-
+const uploadMetadata = async (params: DomainMetadataParams) => {
+	// upload images to http backend
 	const imageResponse = await fetch(uploadApiEndpoint, {
 		method: 'POST',
 		body: JSON.stringify(params.image),
 	});
 	const image = (await imageResponse.json()) as UploadResponseDTO;
 
-	// upload metadata to IPFS
-	const metadataObject = {
+	let metadataObject: MetadataObject = {
 		name: params.name,
 		description: params.story,
-		image: image.url,
+		imageUri: image.url,
 	};
+
+	if (params.previewImage) {
+		//if params has the optional preview image it builds metadata with it
+		const previewImageResponse = await fetch(uploadApiEndpoint, {
```

Comment:
This segment of code:

```
const previewImageResponse = await fetch(uploadApiEndpoint, {
	method: 'POST',
	body: JSON.stringify(params.previewImage), // only difference
});
```

is repeated twice in this method. Consider refactoring this into a helper method.